init.sls

```diff
@@ -106,20 +106,48 @@ chown_pgdata:
 # Deploy users. Sort them, so order is not changing between salt runs
 {% for config in pillar['postgresql']['users']|default({}) %}
 createuser-{{ loop.index }}:
-  cmd.run:
-    - name: psql -t -c "CREATE ROLE {{ config['username'] }} {{ config['options']|default('NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN') }};"
-    - unless: psql -t -c "SELECT 1 FROM pg_roles WHERE rolname='{{ config['username'] }}'" |grep -q 1
-    - runas: postgres
+  postgres_user.present:
+    - name: {{ config['username'] }}
+    - encrypted: True
+    {% if config['login']|default(false) %}
+    - login: True
```

Comment:
I think you could use the pillar value here directly:

```
- login: {{ config['login'] | default(false) %}
```

Reply:
Also we should adhere to the defaults. Login for example defaults to `true`. 
See: https://www.postgresql.org/docs/11/app-createuser.html

Reply:
As I wrote in the initial comment, the ownership behavior is changeing from first defined user, to last defined user, which even updates on append. This is kind of counter intuitiv, right? I am not sure if we should use the ```postgres_database.present``` state in the first place. WHat do you think?

Reply:
There's a Jinja filter called [reverse()](https://jinja.palletsprojects.com/en/2.10.x/templates/#reverse) maybe just use that in the for loop?

Reply:
Yeah, it is kind of irritating, but the ```postgres_user``` state is useing ```CREATE ROLE```, which defaults to NOLOGIN.
https://github.com/saltstack/salt/blob/develop/salt/modules/postgres.py#L1149
https://www.postgresql.org/docs/current/sql-createrole.html

Reply:
I agree that this is confusing.
I think we should default to `true` here, because our pillar entry is called `postgresq:users` and not `postgresql:roles`, and I think we should therefore use the default for `CREATE USER`, at least to me this seems more intuitively.

Alternatively, we could rename everything to `roles`, which might be the more correct approach, but I think we should hold that back for a future release (maybe), what do you think?

Reply:
Yes i agree, that it is more intuitively and more consistent with the strange nameing of the salt state. Basically we are creating an ```alias```, as it is done in ```CREATE USER```.
I guess this is also better, because ```login: true``` is probably the more frequently used option.